tests/modules/users/repository/user.repository.spec.ts

```diff
@@ -0,0 +1,146 @@
+import {
+  createUser,
+  fetchUser,
+  getExistingUser
+} from '@modules/users/repository/user.repository';
+import { User } from '@prisma/client';
+import { prismaMock } from '@tests/prismaTestSetup';
+
+const mockDataStore = [
+  {
+    id: '1',
+    firstName: 'John',
+    lastName: 'Doe',
+    email: 'john.doe@example.com',
+    phoneNumber: '123-456-7890',
+    address: '123 Main St, Anytown USA',
+    isActive: true,
+    isVerified: true,
+    createdAt: new Date(),
+    updatedAt: new Date()
+  },
+  {
+    id: '2',
+    firstName: 'Jane',
+    lastName: 'Doe',
+    email: 'jane.doe@example.com',
+    phoneNumber: '987-654-3210',
+    address: '456 Oak St, Anytown USA',
+    isActive: false,
+    isVerified: true,
+    createdAt: new Date(),
```

Comment:
Let's use date value instead of the new Date(). Date instance could be different from the beginning of test to the actual assertion, which could raise an issue.